File: backend/api/account.py
from fastapi import APIRouter, Depends, Response
from sqlalchemy.orm import Session

from backend.api.deps import get_current_active_user
from backend.db import get_db
from backend.models.user import User
from backend.schemas.user import MeResponse, UpdateAccountRequest

router = APIRouter()


@router.get("/me", response_model=MeResponse)
def get_me(current_user: User = Depends(get_current_active_user)):
    return current_user


@router.put("/account", response_model=MeResponse)
def update_account(
    body: UpdateAccountRequest,
    current_user: User = Depends(get_current_active_user),
    db: Session = Depends(get_db),
):
    if body.notification_advance_minutes is not None:
        current_user.notification_advance_minutes = body.notification_advance_minutes
    db.commit()
    db.refresh(current_user)
    return current_user


@router.delete("/account", status_code=204)
def delete_account(
    current_user: User = Depends(get_current_active_user),
    db: Session = Depends(get_db),
):
    db.delete(current_user)
    db.commit()
    return Response(status_code=204)

File: frontend/src/api/account.ts
import { apiFetch } from './client'
import type { CurrentUser } from '../types'

export function deleteAccount(): Promise<void> {
  return apiFetch<void>('/api/account', { method: 'DELETE' })
}

export function getMe(): Promise<CurrentUser> {
  return apiFetch<CurrentUser>('/api/me')
}

export function updateAccount(data: { notification_advance_minutes?: number }): Promise<CurrentUser> {
  return apiFetch<CurrentUser>('/api/account', {
    method: 'PUT',
    body: JSON.stringify(data),
  })
}
